### ISSUE TITLE
F# compiler doesn't emit default parameter value for some types

### ISSUE THREAD

**Issue by auduchinok** [dotnet/fsharp#16112]
Consider these C# and VB.NET declarations that have default parameter values:
```csharp
public class C1
{
    public static void M(decimal x = 4) => Console.Write(x);
}
```
```vbnet
Public Class C2
    Public Shared Sub M(Optional d As Date = #2018/01/01#)
        Console.WriteLine(d)
    End Sub
End Class
```

F# compiler allows using these methods:
```fsharp
C1.M()
C2.M1()
```

The problem is it doesn't emit the parameter default values inside the calls, and emits type default values instead.
However, it works as expected when other allowed types are used, like `int` or `string`.
Since F# compiler allows skipping arguments for such parameters it seems that emitting the type default value, especially only for some types while working properly for others, might be a bug.

**Comment by auduchinok** [comment]
@vzarytovskii I see, thanks.

It seems that from the user point of view this may be undesirable and I'd expect the most people to not know about the problem when they make such methods calls inside F# code. Would it be possible to change the behaviour so it matches the Roslyn one? Should I open an issue in F# lang suggestions if yes?

**Comment by vzarytovskii** [comment]
> @vzarytovskii I see, thanks.
> 
> It seems that from the user point of view this may be undesirable and I'd expect the most people to not know about the problem when they make such methods calls inside F# code. Would it be possible to change the behaviour so it matches the Roslyn one? Should I open an issue in F# lang suggestions if yes?

Yeah, suggestion should be fine (or even add to existing one, pretty sure there's one).
